### PR TITLE
[SROTEL-232] port out of range:-1

### LINKED ISSUE #231: Getting port out of range:-1 exception when using a default port
If the request goes to a default port (80 / 443) then the request URI port is undefined which causes OpenTelemetryServerFilter#NetServerAttributesExtractor.getHostSocketAddress() to fail

### PR DESCRIPTION
**Pull request by ehsavoie**
* Using the default URL port if no port is defined for the request URI.
 * Fixes #231

### CODE REVIEW

**Comment by ehsavoie** [comment]
Backport PP for 2.3.x: https://github.com/smallrye/smallrye-opentelemetry/pull/232

File: implementation/rest/src/main/java/io/smallrye/opentelemetry/implementation/rest/OpenTelemetryServerFilter.java [hunk 1]
```diff
@@ -141,8 +150,7 @@ protected InetSocketAddress getPeerSocketAddress(final ContainerRequestContext r
 
         @Override
         protected InetSocketAddress getHostSocketAddress(final ContainerRequestContext request) {
-            URI requestUri = request.getUriInfo().getRequestUri();
-            return new InetSocketAddress(requestUri.getHost(), requestUri.getPort());
+            return new InetSocketAddress(getHostName(request), getHostPort(request));
```
Comment (jamezp) [inline implementation/rest/src/main/java/io/smallrye/opentelemetry/implementation/rest/OpenTelemetryServerFilter.java:153]: Should we guard against -1 being returned or are we okay with the `InetSocketAddress` throwing the exception?

Comment (ehsavoie) [reply implementation/rest/src/main/java/io/smallrye/opentelemetry/implementation/rest/OpenTelemetryServerFilter.java:153]: In the case the port can't be found from the scheme then I'd rather have a -1

**Review by xstefank — APPROVED** [review]
LGTM

**Review by radcortez — APPROVED** [review]

**Pull request merged**